### PR TITLE
chore: Resolve all `typedoc` related warnings

### PR DESCRIPTION
**Pull request by IvayloAndonov**
# Context

`deploy-docs` workflow has a lot of [warnings](https://github.com/input-output-hk/cardano-js-sdk/runs/7558744346?check_suite_focus=true#step:4:806) which makes for a lot of missing references (links) in generated docs, e.g. [DbSyncAssetProvider | Documentation](https://input-output-hk.github.io/cardano-js-sdk/classes/_cardano_sdk_cardano_services.DbSyncAssetProvider.html) doesn’t have a link to AssetProvider.

`typedoc` currently complains with a bunch of warnings like this:
```
Warning: Channel, defined at packages/web-extension/src/messaging/BackgroundMessenger.ts:7, is referenced by @cardano-sdk/web-extension.BackgroundMessengerApiDependencies.__type.__type.__type but not included in the documentation.
```

Explore `typedoc` configuration and try to resolve all warnings and issues related to SDK's `typedoc` documentation.
# Proposed Solution

_**Warnings**_ 

**Before**
Screenshot 2023-01-25 at 13 34 28

**After**
Screenshot 2023-01-25 at 13 30 41

**_Docs_** 

**Before** (**DbSyncAssetProvider**)
Screenshot 2023-01-25 at 13 23 38

**After** 
(**DbSyncAssetProvider**)
Screenshot 2023-01-25 at 13 29 24

(**@cardano-services**)
Screenshot 2023-01-25 at 13 29 56
Screenshot 2023-01-25 at 13 41 04

# Important Changes Introduced
- Check the commits messages

### CODE REVIEW

File: typedoc.json [hunk 1]
```diff
@@ -1,5 +1,24 @@
 {
   "out": "./docs",
+  "readme": "none",
+  "exclude": [ "**/node_modules/**"],
   "excludeInternal": true,
-  "readme": "none"
+  "entryPointStrategy": "packages",
+  "entryPoints": [
```
Comment (mkazlauskas) [inline typedoc.json:7]: Please confirm that it doesn't work without this

**Review by mkazlauskas — DISMISSED** [review]
Nice work!

**Review by rhyslbw — CHANGES_REQUESTED** [review]
Commits a37263e9f7d43d782e296021a7bc4322775b1bf0 and 4bdf0a602c8244cc4582b829b88d8cdf9150a857 should be of type `docs`

File: typedoc.json (hunk 1 above)
Comment (IvayloAndonov) [reply typedoc.json:7]: Yeah, without `entrypoints` it produces new warnings, and this one is specific so it doesn't work correctly:
```
warning The --name option was not specified, and no package.json was found
```

**Comment by IvayloAndonov** [comment]
> Commits [a37263e](https://github.com/input-output-hk/cardano-js-sdk/commit/a37263e9f7d43d782e296021a7bc4322775b1bf0) and [4bdf0a6](https://github.com/input-output-hk/cardano-js-sdk/commit/4bdf0a602c8244cc4582b829b88d8cdf9150a857) should be of type `docs`

Commits are squashed and commit types are updated accordingly. @rhyslbw

**Review by rhyslbw — APPROVED** [review]
Great work @IvayloAndonov !

**Pull request merged**